### PR TITLE
As/start tether stop tether

### PR DESCRIPTION
**Pull request by andrew-zenefits**
disable tethering while tooltips are hidden (https://github.com/sir-dunxalot/ember-tooltips/pull/96)

We have also added tests to ensure that the `is-tether-enabled` property aligns with the `isShown` property.

### Before
reposition-before

### After (it repositions if the tooltip is shown; aka when aria-hidden="false")
reposition-after

### CODE REVIEW

**Comment by andrew-zenefits** [comment]
Tests are all passing except `ember-beta`, which is failing with the expected `You must specify a target attribute in the format target="#element-id" for the tooltip component` message which is due to using `this.get('parentView')`, which is deprecated

File: addon/components/tooltip-and-popover.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default EmberTetherComponent.extend({
 
   /* Properties */
 
-  attributeBindings: ['aria-hidden', 'role', 'tabindex'],
+  attributeBindings: ['aria-hidden', 'role', 'tabindex', 'is-tether-enabled'],
```
Comment (andrew-zenefits) [inline addon/components/tooltip-and-popover.js]: is it OK to add a random attribute here? This doesn't affect the visuals or behavior of the component. We only add it so we can test it during the `assertShow` and `assertHide` functions

Comment (jliuzen) [inline addon/components/tooltip-and-popover.js]: does it make sense to update the attribute name to indicate it's for testing purposes (similar to our `js-glue-*` convention for class names)?

**Comment by fhuzenefits** [comment]
I did some profiling about ember-tooltip. This PR will help improve performance. But for a complicated dynamic page, ember-tooltip still consumes a lot of cpu resources. 

Here is our PR, `https://github.com/zenefits/yourPeople3/pull/15502/files` for a page with a few ember-tooltip in it. 
The action I did for profiling is to toggle a button to render a bunch of components with ember-tooltip inside. UI is still freezing for a couple of seconds. The experiment was done with this branch of ember-tooltip. 
image

Not sure if we can find a workaround to solve the problem. I guess we might need to get rid of `ember-tooltip` for this page and look for some other solutions.

@roman-tereshin ^

File: addon/components/tooltip-and-popover.js (hunk 1 above)
Comment (a15n) [inline addon/components/tooltip-and-popover.js]: I think that might be slightly better but I don't think it's necessary. This is just a patch till the original lib supports this. I plan on making a PR to the lib and suggest that we prefix everything Javascript related with a `js-` and will update then

File: tests/integration/components/tooltip-is-visible-test.js [hunk 2]
```diff
@@ -11,18 +11,18 @@ moduleForComponent('tooltip-on-component', 'Integration | Option | isShown', {
...
-  assert.expect(2);
+  assert.expect(4);
 
-  this.set('showTooltip', true);
+  this.set('showTooltip', false);
 
   this.render(hbs`{{tooltip-on-component isShown=showTooltip}}`);
 
-  assertShow(assert, this);
+  assertHide(assert, this);
```
Comment (andrew-zenefits) [inline tests/integration/components/tooltip-is-visible-test.js]: I should not have made this change. I have reverted it here...

https://github.com/zenefits/ember-tooltips/pull/24

**Pull request merged**